packages/flutter/lib/src/material/input_decorator.dart

```diff
@@ -2,22 +2,1623 @@
+      + _maxWidth(suffix, height)
+      + _maxWidth(suffixIcon, height)
+      + contentPadding.right;
+  }
+
+  double _lineHeight(double width, List<RenderBox> boxes) {
+    double height = 0.0;
+    for (RenderBox box in boxes) {
+      if (box == null)
+        continue;
+      height = math.max(_minHeight(box, width), height);
+    }
+    return height;
+    // TODO(hansmuller): this should compute the overall line height for the
+    // boxes when they've been baseline-aligned.
+    // See https://github.com/flutter/flutter/issues/13715
+  }
+
+  @override
+  double computeMinIntrinsicHeight(double width) {
+    double subtextHeight = _lineHeight(width, <RenderBox>[helperError, counter]);
+    if (subtextHeight > 0.0)
+      subtextHeight += 8.0;
+    return contentPadding.top
+      + (label == null ? 0.0 : decoration.floatingLabelHeight)
+      + _lineHeight(width, <RenderBox>[prefix, input, suffix])
+      + subtextHeight
+      + contentPadding.bottom;
+  }
+
+  @override
+  double computeMaxIntrinsicHeight(double width) {
+    return computeMinIntrinsicHeight(width);
+  }
+
+  @override
+  double computeDistanceToActualBaseline(TextBaseline baseline) {
+    // TODO(hansmuller): layout first? Add input.parentData.offset?
+    assert(false);
+    return 0.0;
+  }
+
+  // Records where the label was painted.
+  Matrix4 _labelTransform;
+
+  @override
+  void performLayout() {
+    _labelTransform = null;
+    final _RenderDecorationLayout layout = _layout(constraints);
+
+    final double overallWidth = constraints.maxWidth;
+    final double overallHeight = layout.containerHeight + layout.subtextHeight;
+
+    if (container != null) {
+      final BoxConstraints containerConstraints = new BoxConstraints.tightFor(
+        height: layout.containerHeight,
+        width: overallWidth - _boxSize(icon).width,
+      );
+      container.layout(containerConstraints, parentUsesSize: true);
+      final double x = textDirection == TextDirection.rtl ? 0.0 : _boxSize(icon).width;
+      _boxParentData(container).offset = new Offset(x, 0.0);
+    }
+
+    double height;
+    double centerLayout(RenderBox box, double x) {
+      _boxParentData(box).offset = new Offset(x, (height - box.size.height) / 2.0);
+      return box.size.width;
+    }
+
+    double baseline;
+    double baselineLayout(RenderBox box, double x) {
+      _boxParentData(box).offset = new Offset(x, baseline - layout.boxToBaseline[box]);
+      return box.size.width;
+    }
+
+    final double left = contentPadding.left;
+    final double right = overallWidth - contentPadding.right;
+
+    height = layout.containerHeight;
+    baseline = decoration.border is OutlineInputBorder
+      ? layout.outlineBaseline
+      : layout.inputBaseline;
+
+    if (icon != null) {
+      final double x = textDirection == TextDirection.rtl ? overallWidth - icon.size.width : 0.0;
+      centerLayout(icon, x);
+    }
+
+    if (textDirection == TextDirection.rtl) {
```

Comment:
prefer `switch` instead of `if` when checking an enum value

Reply:
done